internal/v1/no_auth/api_no_auth.go

```diff
@@ -0,0 +1,219 @@
+	Rhel86       Distributions = "rhel-86"
+	Rhel87       Distributions = "rhel-87"
+	Rhel88       Distributions = "rhel-88"
+	Rhel89       Distributions = "rhel-89"
+	Rhel8Nightly Distributions = "rhel-8-nightly"
+	Rhel9        Distributions = "rhel-9"
+	Rhel90       Distributions = "rhel-90"
+	Rhel91       Distributions = "rhel-91"
+	Rhel92       Distributions = "rhel-92"
+	Rhel93       Distributions = "rhel-93"
+	Rhel9Nightly Distributions = "rhel-9-nightly"
+)
+
+// DistributionProfileItem defines model for DistributionProfileItem.
+type DistributionProfileItem string
+
+// Distributions List of all distributions that image builder supports. A user might not have access to
+// restricted distributions.
+//
+// Restricted distributions include the RHEL nightlies and the Fedora distributions.
+type Distributions string
+
+// ServerInterface represents all server handlers.
+type ServerInterface interface {
+	// get the openapi json specification
+	// (GET /openapi.json)
+	GetOpenapiJson(ctx echo.Context) error
+}
+
+// ServerInterfaceWrapper converts echo contexts to parameters.
+type ServerInterfaceWrapper struct {
+	Handler ServerInterface
+}
+
+// GetOpenapiJson converts echo context to params.
+func (w *ServerInterfaceWrapper) GetOpenapiJson(ctx echo.Context) error {
+	var err error
+
+	// Invoke the callback with all the unmarshalled arguments
+	err = w.Handler.GetOpenapiJson(ctx)
+	return err
+}
+
+// This is a simple interface which specifies echo.Route addition functions which
+// are present on both echo.Echo and echo.Group, since we want to allow using
+// either of them for path registration
+type EchoRouter interface {
+	CONNECT(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	DELETE(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	GET(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	HEAD(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	OPTIONS(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	PATCH(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	POST(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	PUT(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+	TRACE(path string, h echo.HandlerFunc, m ...echo.MiddlewareFunc) *echo.Route
+}
+
+// RegisterHandlers adds each server route to the EchoRouter.
+func RegisterHandlers(router EchoRouter, si ServerInterface) {
+	RegisterHandlersWithBaseURL(router, si, "")
+}
+
+// Registers handlers, and prepends BaseURL to the paths, so that the paths
+// can be served under a prefix.
+func RegisterHandlersWithBaseURL(router EchoRouter, si ServerInterface, baseURL string) {
+
+	wrapper := ServerInterfaceWrapper{
+		Handler: si,
+	}
+
+	router.GET(baseURL+"/openapi.json", wrapper.GetOpenapiJson)
+
+}
+
+// Base64 encoded, gzipped, json marshaled Swagger object
+var swaggerSpec = []string{
```

Comment:
We've moved away from inlining the spec in https://github.com/osbuild/image-builder/pull/1012

Reply:
... sorry - copy'n'paste :-/